### ISSUE TITLE
How to draw a 3D model composed of non-triangles?

### ISSUE THREAD

**Issue by vkter** [plotly/plotly.py#3557]
In matplotlib, I can use "add_collection3d()" to draw

What should I do in Ploty?

**Comment by vkter** [comment]
The model data is made as follows:
v:[[1,1,1],[1,3,2],[2,3,2],...]
f:[[0,1,2,3],[6,7,8],[8,7,6,,9],...]
Each surface data consists of a vertex index constituting it

**Comment by empet** [comment]
Your mesh consists in the array of vertices, and faces, reresented by qudrangles (rectangles). To be able to define a `go. Mesh3d` you must triangulate the quadrangles .
Let us say you have read the mesh and
`verts` is the numpy array of vertices and `quads` the array fo quadrangles. Then you get the list of triangles as follows:
```
triangles = []
for r in quads:
    triangles. extend([[r[0], r[1], r[2]], [r[0], r[2], r[3]]] # split each rectangle in two triangles
triangles = np.asarray(triangles)
I, J, K = triangles.T
x, y, z = verts.T
fig=go.Figure(go.Mesh3d(x=x, y=y, z=z, i=I, j=J, k=K, colorscale="matter_r", intensity=z)
```

**Comment by vkter** [comment]
The face may not only quadrilateral, and there are too many surface data, and manual transformation is unlikely.

Does Ploty can consider supporting polygonal data

**Comment by empet** [comment]
No, Plotly does not suport rendering  polygonal data directly. But the  convex polygonal faces can be  triangulated before instantiating  `go.Mesh3d`. Any  graphical library proceeds at polygonal triangulation before rendering, performs  boolean  operations on the faces, edges  To use Plotly, you must  perform yourself such operations. 
Each 𝑛-polygonal face, `𝑛>3`, represented by a list of vertex indices, `[i0, i1, ... in-1]`, is decomposed into 𝑛−2 triangular faces:
[`i𝟶, i𝚔, i𝚔+𝟷]`, with `𝑘=1,…,𝑛−2`, where `ik`  is LaTeXed to `$i_k$`
